gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenAuthorityService.java

```diff
@@ -78,17 +83,22 @@
   private GatewayConfig config;
 
   private char[] cachedSigningKeyPassphrase;
+  private byte[] cachedSigningHmacSecret;
   private RSAPrivateKey signingKey;
 
   static {
-      // Only standard RSA signature algorithms are accepted
+      // Only standard RSA and HMAC signature algorithms are accepted
       // https://tools.ietf.org/html/rfc7518
       SUPPORTED_SIG_ALGS.add("RS256");
       SUPPORTED_SIG_ALGS.add("RS384");
       SUPPORTED_SIG_ALGS.add("RS512");
       SUPPORTED_SIG_ALGS.add("PS256");
       SUPPORTED_SIG_ALGS.add("PS384");
       SUPPORTED_SIG_ALGS.add("PS512");
+      //HMAC
+      SUPPORTED_SIG_ALGS.add("HS256");
+      SUPPORTED_SIG_ALGS.add("HS384");
+      SUPPORTED_SIG_ALGS.add("HS512");
```

Comment:
Would it make sense to have two `SUPPORTED_SIG_ALGS` lists? One for PKI and the other for HMAC? Then can check and hopefully provide a better error message if the wrong signature algorithm is chosen for the wrong method?

Reply:
👍

Reply:
Done and tested:
```
2021-01-12 10:32:35,717 ERROR service.knoxtoken (TokenResource.java:getAuthenticationToken(439)) - Unable to issue token.
org.apache.knox.gateway.services.security.token.TokenServiceException: The Knox Gateway is configured to use PKI as signing method, however the supplied algorithm (HS256) is incompatible with that method
	at org.apache.knox.gateway.services.token.impl.DefaultTokenAuthorityService.signToken(DefaultTokenAuthorityService.java:175)
	at org.apache.knox.gateway.services.token.impl.DefaultTokenAuthorityService.issueToken(DefaultTokenAuthorityService.java:157)
	at org.apache.knox.gateway.services.token.impl.DefaultTokenAuthorityService.issueToken(DefaultTokenAuthorityService.java:137)
	at org.apache.knox.gateway.service.knoxtoken.TokenResource.getAuthenticationToken(TokenResource.java:401)
	at org.apache.knox.gateway.service.knoxtoken.TokenResource.doGet(TokenResource.java:233)
```